lib/iris/coords.py

```diff
@@ -1611,7 +1612,11 @@ def _sanitise_array(self, src, ndmin):
     def points(self):
         """Property containing the points values as a numpy array"""
         if is_lazy_data(self._points):
-            self._points = self._points.compute()
+            self._points = as_concrete_data(self._points,
+                                            nans_replacement=np.ma.masked)
+            # NOTE: we probably don't have full support for masked aux-coords.
+            # We certainly *don't* handle a _FillValue attribute (and possibly
+            # the loader will throw one away ?)
```

Comment:
@bjlittle I've just raised https://github.com/SciTools/iris/issues/2449
 not sure if that's quite what you were after. Feel free to edit the issue